src/utils.ts

```diff
@@ -0,0 +1,235 @@
+export const get_from_dict = (obj: object, key: string): any => {
+    let keyParts: string[] = String(key).split('.');
```

Comment:
Would be better as:

`const keyParts = key.split(".");`

`key` is already a string, and TypeScript knows that String.split returns string[].

Extra type annotations just make things look more confusing IMHO.